src/api.ts

```diff
@@ -63,6 +63,16 @@ export class Api {
       headers: { "Content-Type": "application/json" },
     });
   }
+
+  async getExecutedSellAmount(uid: string): Promise<BigNumber> {
+    const response: OrderDetailResponse = await this.call(`orders/${uid}`);
+    return BigNumber.from(response.executedSellAmount);
+  }
```

Comment:
Doesn't this depend on the services having received and stored the event? If I understand the purpose of this PR correctly we are trying to eliminate the issue of not having received the event before claiming failure. What if the services also didn't get the event in time?

Reply:
The database has a more sophisticated polling approach to fetch events (which is what Nick alluded to implement here as well in his comment but which I think goes beyond the scope of what's reasonable for this bot). 

As far as this bot is concerned it should assume the API has a means to get this information reliably.